ast/edit/option_editor.go

```diff
@@ -0,0 +1,141 @@
+package edit
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+	"github.com/pkg/errors"
+)
+
+var OPT_NOT_FOUND = errors.New("option not found")
+
+type EditFn func(opt *ast.OptionStatement) (ast.Expression, error)
+
+/*
+`Edit` passes the `OptionStatement` in the AST rooted at `node` that has the
+specified identifier to `editFn`.
+The function can have side effects on the option statement
+and/or return a non-nil `Expression` that is set as value for the option.
+If the value returned by the edit function is `nil` (or an error is returned) no new value is set
+for the option statement (but any, maybe partial, side effect is applied).
+If the option is not present in the AST, an OPT_NOT_FOUND error is returned.
+*/
+func Edit(node ast.Node, optionIdentifier string, editFn EditFn) error {
+	oe := &optionEditor{identifier: optionIdentifier, editFn: editFn, err: nil}
+	ast.Walk(oe, node)
+
+	if oe.err != nil {
+		return oe.err
+	}
+
+	if !oe.found {
+		return OPT_NOT_FOUND
+	}
+
+	return nil
+}
+
+// Handy function for setting the value of an `OptionStatement`.
+func EditValue(node ast.Node, optionIdentifier string, expr ast.Expression) error {
+	editFn := func(opt *ast.OptionStatement) (ast.Expression, error) {
+		return expr, nil
+	}
+	return Edit(node, optionIdentifier, editFn)
+}
+
+/*
+Handy function for updating the values of an `OptionStatement` that has an
+`ObjectExpression` as value.
+*/
+func EditObject(node ast.Node, optionIdentifier string, keyMap map[string]values.Value) error {
+	editFn := func(opt *ast.OptionStatement) (ast.Expression, error) {
+		obj, ok := opt.Assignment.Init.(*ast.ObjectExpression)
+		if !ok {
+			return nil, fmt.Errorf("value is not an object expression, it is %s instead", opt.Assignment.Init.Type())
+		}
+
+		for _, p := range obj.Properties {
+			value, found := keyMap[p.Key.Name]
+			if found {
+				lit, err := GetLiteral(value)
+				if err != nil {
+					return nil, err
+				}
+
+				p.Value = lit
+			}
+		}
+
+		return nil, nil
+	}
+	return Edit(node, optionIdentifier, editFn)
```

Comment:
The EditObject function should return an EditFn instead of calling Edit directly. Then it can easily be reused.

Same for the EditValue function.